src/shared/clients/codecatalystClient.ts

```diff
@@ -729,23 +745,17 @@ class CodeCatalystClientInternal {
         )
 
         const devenv = await waitTimeout(pollDevEnv, timeout).catch(e => {
-            const err = e as Error
-            const starts = statuses.filter(o => o.status === 'STARTING').length
-            const fails = statuses.filter(o => o.status === 'FAILED').length
-
-            if (!isUserCancelledError(e)) {
-                doLog('error', 'devenv failed to start')
-            } else {
+            if (isUserCancelledError(e)) {
                 doLog('info', 'devenv failed to start (user cancelled)')
-                err.message = failedStartMsg()
-                throw err
+                e.message = failedStartMsg()
+                throw e
+            } else if (e instanceof ToolkitError) {
+                doLog('error', 'devenv failed to start')
+                throw e
             }
```

Comment:
pass through all `ToolkitError`s instead of making decisions in this `waitTimeout()` catch.